File: app/src/main/java/com/sachtech/datingapp/ui/home/adapter/MyPagerAdapter.java
package com.sachtech.datingapp.ui.home.adapter;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import androidx.viewpager.widget.PagerAdapter;

import com.sachtech.datingapp.R;
import com.squareup.picasso.Picasso;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

import kotlin.TypeCastException;


public final class MyPagerAdapter extends PagerAdapter {
   @Nullable
   private final Activity context;
   @NotNull
   private final ArrayList arrayList;
   @NotNull
   private LayoutInflater mLayoutInflater;

   @Nullable
   public final Activity getContext() {
      return this.context;
   }


   @NotNull
   public final LayoutInflater getMLayoutInflater$app() {
      return this.mLayoutInflater;
   }



   public boolean isViewFromObject(@NotNull View p0, @NotNull Object p1) {
      return p0 == (RelativeLayout)p1;
   }

   public int getCount() {
      return arrayList.size();
   }

   public void destroyItem(@NotNull ViewGroup container, int position, @NotNull Object object) {
      container.removeView((View)((RelativeLayout)object));
   }

   @NotNull
   public Object instantiateItem(@NotNull ViewGroup container, int position) {
      View itemView = this.mLayoutInflater.inflate(R.layout.pager_item, container, false);
      ImageView var10000 = itemView.findViewById(R.id.imageView);
         Picasso.get().load((String)this.arrayList.get(position)).into(var10000);
         container.addView(itemView);
         return itemView;
   }

   public MyPagerAdapter(@Nullable Activity activity, @NotNull ArrayList arraylist) {
      super();
      this.context = activity;
      this.arrayList = arraylist;
      Object var10001 = activity != null ? activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE) : null;
      if (var10001 == null) {
         throw new TypeCastException("null cannot be cast to non-null type android.view.LayoutInflater");
      } else {
         this.mLayoutInflater = (LayoutInflater)var10001;
      }
   }
}

File: app/src/main/java/com/sachtech/datingapp/ui/home/adapter/MyPagerAdapter.kt
/*
package com.sachtech.datingapp.ui.home.adapter

import android.app.Activity
import android.content.Context
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import androidx.viewpager.widget.PagerAdapter
import com.sachtech.datingapp.R
import com.sachtech.datingapp.utils.loadImage
import com.squareup.picasso.Picasso

class MyPagerAdapter(activity: Activity?, arraylist: ArrayList<String>) : PagerAdapter() {
    val context = activity
    val arrayList = arraylist
    internal var mLayoutInflater: LayoutInflater =
        activity?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    override fun isViewFromObject(p0: View, p1: Any): Boolean {
        return p0 === p1 as RelativeLayout
    }

    override fun getCount(): Int {
        return arrayList.size
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as RelativeLayout)
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val itemView = mLayoutInflater.inflate(R.layout.pager_item, container, false)
        val imageView = itemView.findViewById(R.id.imageView) as ImageView
        Picasso.get().load(arrayList[position]).into(imageView)
        //imageView.setImageURI(Uri.parse(arrayList[position]))
        container.addView(itemView)
        return itemView
    }
}*/
